lib/server.js

```diff
@@ -28,21 +45,76 @@ app.get('/people/:id', (req, res, next) => {
   res.json(record[0]);
 });
 
+/**
+ * @route POST /api/v1/:model
+ * Model must be a proper model, located within the ../models folder
+ * @param {model} model.path.required
+ * @returns {object} 200 - Count of results with an array of results
+ * @returns {Error}  500 - Unexpected error
+ */
+
 // Route to Create a person
 app.post('/people', (req, res, next) => {
   let record = req.body;
-  // what can you do to ensure there are no
-  // duplicate ids?
-  record.id = Math.random();
+  record.id = db.teams.length+=100;
   db.people.push(record);
   res.json(record);
 });
 
 // Route to Update a person
-app.put('/people/:id', (req, res, next) => {});
+app.put('/people/:id', (req, res, next) => {
+  let id = parseInt(req.params.id);
+  let record = db.people.filter(record => record.id === id);
+  record[0].firstName = req.body.firstName;
+  record[0].lastName = req.body.lastName;
+  record[0].birthday = req.body.birthday;
+  record[0].team = req.body.team;   
+  record[0].likes = req.body.likes;
```

Comment:
This is a fine approach, but keep in mind that a `PUT` operation can specify any data to update. So having a body that looks like:
```
{ firstName = "NewName" } 
```
would still be valid, and should only update the first name. Based on the code you're using here, if given the above body, the `PUT` route would actually delete content from `lastName`, `birthday`, `likes` and `team`. A better way to do this would be to check if a key value exists in `req.body` and then edit the record based on which keys are present. Or, even better, you can do this very easily with the spread operator:

```
record = {...record, ...req.body}
```